### DISCUSSION TITLE
Does walletconnect sdk support import new token flow for connected wallets like MM, TW?

### DISCUSSION THREAD

**Discussion by faaqi** [WalletConnect/WalletConnectFlutterV2#313] [Q&A]
Usecase :

- User can search the token by name or contract address for external connected wallet and trigger the import token flow

**Comment by quetool** [comment]
Hello @faaqi! I would say as long as the wallet supports `wallet_watchAsset` then it's supported. After connecting you can check inside session's approved namespaces if this method is supported by the wallet. WalletConnect just provide a secure channel between Dapp and Wallet, we don't determine which rpc methods are going to be supported, we only require wallets to register their supported methods in the session.
